### PR TITLE
Generate Timetable from TaskList and TaskFilter

### PR DESCRIPTION
**Pull request by daniellimws**
#### Summary of changes made to non-Timetable classes:
* `ImportanceFilter`: Create this class to filter tasks based on importance
* `TaskList`: Change `getFilteredTasks` to use `TaskFilter`
* `Task`: Change` importance` field type to `Importance` enum instead of `String`. Move `Task` class into `athena.task` package.
* `Parser` and `Command` classes: Update methods to use `Importance` enum
* `ListCommand`: Generate a timetable to be printed using `ui.printTimetable`

(Sorry all these were side effects due to using an `ImportanceFilter` class for filtering tasks)

* `Ui`: Uncomment `printTimetable`

#### Changes made to classes in `Timetable` package:
* Implement `toString()` methods for `Timetable` and `TimetableDay`
* `Timetable` can generate a list of `TimetableDay` objects from a `TaskList` and an optional `TaskFilter`.

Because of the way dates are parsed in `Timetable`, now we only can accept dates/recurrences (once off recurrence) in the `dd-MM-yyyy` format (Monday, Tuesday, ... not supported for now). The `LocalDate` class expects a year to be provided so I think we need to help the user add in the year in `Parser`. After this PR, I'll work on accepting days of the week (e.g. Monday, Tuesday, ...) for the recurrence instead of only allowing dates.

#### Tests
Also add tests to make sure `Timetable` correctly generates a list of `TimetableDay` objects from a `TaskList`, as well as from a `TaskList` filtered with an `ImportanceFilter` set to `HIGH` importance.

### CODE REVIEW

**Review by amanda-chua — APPROVED** [review]
looks good

File: src/main/java/athena/Parser.java [hunk 1]
```diff
@@ -123,8 +124,9 @@ public static Command parseEditCommand(String taskInfo, int namePos, int timePos
      */
     public static Command parseListCommand(String taskInfo, int importancePos) {
         String nullValue = "";
-        String importance = getParameterDesc(taskInfo, IMPORTANCE_DELIMITER, importancePos, nullValue);
-        Command command = new ListCommand(importance);
+        String importance = getParameterDesc(taskInfo, IMPORTANCE_DELIMITER, importancePos,
+                Importance.MEDIUM.toString());
+        Command command = new ListCommand(Importance.valueOf(importance.toUpperCase()));
```
Comment (wish2023) [inline src/main/java/athena/Parser.java]: Good that the importance is being converted to upper case here!

File: src/main/java/athena/Athena.java [hunk 2]
```diff
@@ -16,13 +16,14 @@ public static void main(String[] args) {
...
+        Command command = parser.parse("add n/Assignment2 t/1100 d/16-09 D/2 r/12-10-2020 i/high "
+                + "a/Refer to lecture notes");
         command.execute(taskList, athena.ui);
         for (int i = 0; i < 10; i++) {
-            command = parser.parse("add n/Assignment2 t/1100 d/16-09 D/2 r/Monday i/high a/Refer to lecture notes");
+            command = parser.parse("add n/Assignment2 t/1100 d/16-09 D/2 r/12-10-2020 i/high a/Refer to lecture notes");
             command.execute(taskList, athena.ui);
         }
-        command = parser.parse("list");
+        command = parser.parse("list i/HIGH");
```
Comment (wish2023) [inline src/main/java/athena/Athena.java:26]: Remember that the user will type "high" instead of "HIGH"

File: src/main/java/athena/TaskList.java [hunk 3]
```diff
@@ -75,14 +76,14 @@ public Task markTaskAsDone(int taskNumber) {
...
 
         Task task = createTask(name, startTime, duration, deadline, recurrence, importance, notes, index);
         tasks.add(task);
     }
 
     public void addTask(String name, String startTime, String duration,
-                        String deadline, String recurrence, String importance, String notes) {
+                        String deadline, String recurrence, Importance importance, String notes) {
         this.maxIndex += 1;
         int index = maxIndex;
```
Comment (wish2023) [inline src/main/java/athena/TaskList.java]: I think we can remove the first addTask() . The method right below this seems to have the same functionality in addition to assigning a unique index value

File: src/main/java/athena/commands/ListCommand.java [hunk 4]
```diff
@@ -22,7 +26,8 @@ public ListCommand(String importance) {
      */
     @Override
     public void execute(TaskList taskList, Ui ui) {
-        TaskList tasksFiltered = taskList.getFilteredList(taskImportance);
-        ui.printList(tasksFiltered.getTasks());
+        ImportanceFilter importanceFilter = new ImportanceFilter(taskImportance);
+        Timetable timetable = new Timetable(taskList, importanceFilter);
+        ui.printTimetable(timetable);
```
Comment (wish2023) [inline src/main/java/athena/commands/ListCommand.java:33]: Noted that the user will now see the list from the timetable class only

File: src/main/java/athena/task/taskfilter/ImportanceFilter.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
+package athena.task.taskfilter;
+
+import athena.Importance;
+import athena.task.Task;
+
+public class ImportanceFilter extends TaskFilter {
```
Comment (wish2023) [inline src/main/java/athena/task/taskfilter/ImportanceFilter.java:6]: Looks great!

**Review by wish2023 — APPROVED** [review]
Looks great! It may be better for the task class to have the deadline and recurrence to make it easier for the timetable class. Let's talk about this during the meeting. 

It may be good if one of us make a recurrence class too since recurrence can be a day of the week, or a string.

File: src/main/java/athena/Parser.java (hunk 1 above)
Comment (daniellimws) [reply src/main/java/athena/Parser.java]: Yea I tried without and the program might crash if the user doesn't give it in upper case.

File: src/main/java/athena/TaskList.java (hunk 3 above)
Comment (daniellimws) [reply src/main/java/athena/TaskList.java]: I agree!

**Comment by daniellimws** [comment]
Thanks. I agree it would be better to have a `Recurrence` class so that we can support different types of values.

File: src/main/java/athena/TaskList.java (hunk 3 above)
Comment (daniellimws) [reply src/main/java/athena/TaskList.java]: Oh wait, one allows the programmer to choose the index while the other will automatically assign an index. But yea there's repeated code so I'll modify the second `addTask` to call the first `addTask`.

**Pull request merged**